### PR TITLE
Nuke: improving farm publishing with workflow

### PR DESCRIPTION
**Pull request by jakubjezek001**
## Changelog Description
Implementing new farm publishing abstractions. Also fixing existing frames validation.

## Testing notes:
1. all publishing targets should be working as expected from `render` and `prerender` family.

## Dependencies
Changes from this PR were initially implemented in [this pull request](https://github.com/ynput/OpenPype/pull/5451). However, due to its significant size, we made the decision to divide it into smaller portions. 

This PR is based on #5498

**!!! This PR should be merged after #5498. But before the merge the target branch should be changed to develop !!!**

### CODE REVIEW

File: openpype/hosts/houdini/plugins/publish/collect_karma_rop.py [hunk 1]
```diff
@@ -31,31 +32,33 @@ class CollectKarmaROPRenderProducts(pyblish.api.InstancePlugin):
...
+        files_by_aov = {
+            "beauty": file_collection.generate_expected_filepaths(
+                frame_start, frame_end, beauty_product)
+        }
 
-            filenames = list(render_products)
-            instance.data["files"] = filenames
-            instance.data["renderProducts"] = colorspace.ARenderProduct()
+        filenames = list(render_products)
+        instance.data["files"] = filenames
```
Comment (Minkiu) [inline openpype/hosts/houdini/plugins/publish/collect_karma_rop.py:60]: We don't seem to be using `filenames` anywhere else so I'd say: `instance.data["files"] = list(render_products)`

File: openpype/pipeline/colorspace.py [hunk 2]
```diff
@@ -13,12 +13,17 @@
...
 from openpype.pipeline import Anatomy
+from openpype.lib.transcoding import VIDEO_EXTENSIONS, IMAGE_EXTENSIONS
+
 
 log = Logger.get_logger(__name__)
 
 
-class CashedData:
+class CachedData:
     remapping = None
```
Comment (Minkiu) [inline openpype/pipeline/colorspace.py:23]: AS Roy commented in a similar PR, this should be a `dict` instead of `None`

**Comment by tokejepsen** [comment]
This is a big PR that touches on other hosts than Nuke. Could we either split this into smaller PRs per host or get a check list of hosts to test?

Also got merge conflicts to resolve before further testing.

**Comment by mkolar** [comment]
Because we're splitting OpenPype into ayon-core and individual host addons, this PR would have to be re-created to target one of those.

We're closing it down, but we'll he happy for a new PR to [ynput/ayon-core](https://github.com/ynput/ayon-core) or the host addon repository once it's up.

**Pull request closed without merging**